tests/shape/test-shape.lua

```diff
@@ -0,0 +1,36 @@
+-- Test if shape crash when called
+-- Also generate some SVG to be used by the documentation
+-- it also "prove" that the code examples are all working
+local cairo = require( "lgi"         ).cairo
+local shape = require( "gears.shape" )
+local filepath, svgpath = ...
+
+-- This is normal when running test, it will run twice
+if (not filepath) or (not svgpath) or key then
+    os.exit(0)
```

Comment:
I will have to figure out how to write tests that run on both the test run and the pre-processing run. One give coverage and the other the documentation. This will be improved in future commits.